### ISSUE TITLE
Cmake and CXX_STANDARD properties are not working anymore

### ISSUE THREAD

**Issue by Lectem** [emscripten-core/emscripten#9100]
It seems emscipten toolchain files are not working with the CXX_STANDARD properties anymore.
This had been fixed in #4571 (issue was #4129) and should be covered by test https://github.com/emscripten-core/emscripten/blob/master/tests/cmake/stdproperty/CMakeLists.txt, but it seems it is not enough ?

Here are the versions information:
```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.38.40
clang version 6.0.1 (Cswircachegitchromium.googlesource.com-external-github.com-emscripten--core-emscripten--fastcomp--clang 98df4be387dde3e3918fa5bbb5fc43e1a0e1daac) (Cswircachegitchromium.googlesource.com-external-github.com-emscripten--core-emscripten--fastcomp 1b4148f39a69c7fc62edadd85e4122b68694dfb7) (emscripten 1.38.31 : 1.38.31)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\emsdk\fastcomp\fastcomp\bin
shared:INFO: (Emscripten: Running sanity checks)
```
Cmake version is 3.15.0

Here is a simple test to reproduce:

# CMakeLists.txt

```cmake
cmake_minimum_required(VERSION 3.4)
project(helloworld)

add_executable(helloworld main.cpp)

set_property(TARGET helloworld PROPERTY CXX_STANDARD_REQUIRED ON)
set_property(TARGET helloworld PROPERTY CXX_STANDARD 11)
set(CMAKE_EXECUTABLE_SUFFIX_CXX .html)
```
# main.cpp

```cpp
struct vector2d{
	int x,y,z;
};

int main(int argc, char * argv[])
{
	vector2d v{0,0,0};
}
```

Which results to the following error:

> E:\progs\WEB\cmakeemscriptencxx11\build>emcmake cmake ..
> -- Configuring done
> -- Generating done
> -- Build files have been written to: E:/progs/WEB/cmakeemscriptencxx11/build
> 
> E:\progs\WEB\cmakeemscriptencxx11\build>ninja
> [1/2] Building CXX object CMakeFiles/helloworld.dir/main.cpp.o
> FAILED: CMakeFiles/helloworld.dir/main.cpp.o
> C:\emsdk\fastcomp\emscripten\em++.bat     -MD -MT CMakeFiles/helloworld.dir/main.cpp.o -MF CMakeFiles\helloworld.dir\main.cpp.o.d -o CMakeFiles/helloworld.dir/main.cpp.o -c ../main.cpp
> ../main.cpp:8:12: error: expected ';' at end of declaration
>         vector2d v{0,0,0};
>                   ^
>                   ;
> 1 error generated.

Note that I also tried MSBuild and nmake buildsystems.

**Comment by kripken** [comment]
This should be fixed now, by #9102 - please confirm @Lectem

**Comment by Lectem** [comment]
Awesome thanks @waywardmonkeys and @kripken , this seems to work perfectly!
I didn't expect a CMake update to break this without any mention in release notes.

I also opened https://gitlab.kitware.com/cmake/cmake/issues/19539 to know more about this issue.
